osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -0,0 +1,364 @@
+        {
+            RelativeSizeAxes = Axes.X,
+            Margin = new MarginPadding { Top = 5 }
+        };
+
+        private sealed class SeedControl : CompositeDrawable, IHasCurrentValue<int>
+        {
+            private readonly BindableWithCurrent<int> current = new BindableWithCurrent<int>();
+
+            public Bindable<int> Current
+            {
+                get => current;
+                set => Scheduler.Add(() => current.Current = value);
+            }
+
+            private readonly OsuNumberBox seedNumberBox;
+
+            public SeedControl()
+            {
+                AutoSizeAxes = Axes.Y;
+
+                InternalChildren = new[]
+                {
+                    new GridContainer
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        AutoSizeAxes = Axes.Y,
+                        ColumnDimensions = new[]
+                        {
+                            new Dimension(),
+                            new Dimension(GridSizeMode.Absolute, 2),
+                            new Dimension(GridSizeMode.Relative, 0.25f)
+                        },
+                        RowDimensions = new[]
+                        {
+                            new Dimension(GridSizeMode.AutoSize)
+                        },
+                        Content = new[]
+                        {
+                            new Drawable[]
+                            {
+                                seedNumberBox = new OsuNumberBox
+                                {
+                                    RelativeSizeAxes = Axes.X,
+                                    CommitOnFocusLost = true
+                                },
+                                null,
+                                new TriangleButton
+                                {
+                                    RelativeSizeAxes = Axes.Both,
+                                    Height = 1,
+                                    Text = "Copy",
+                                    Action = copySeedToClipboard
+                                }
+                            }
+                        }
+                    }
+                };
+
+                seedNumberBox.Current.BindValueChanged(onTextBoxValueChanged);
+            }
+
+            private void onTextBoxValueChanged(ValueChangedEvent<string> e)
+            {
+                string seed = e.NewValue;
+
+                while (!string.IsNullOrEmpty(seed) && !int.TryParse(seed, out _))
+                    seed = seed[..^1];
+
+                if (!int.TryParse(seed, out var intVal))
+                    intVal = 0;
+
+                current.Value = intVal;
+            }
+
+            private void copySeedToClipboard() => host.GetClipboard().SetText(seedNumberBox.Text);
+
+            protected override void Update()
+            {
+                seedNumberBox.ReadOnly = OsuModRandom.CustomSeedDisabled;
```

Comment:
```suggestion
                seedNumberBox.Current.Disabled = OsuModRandom.CustomSeedDisabled;
```

looks better. It is very hard to tell by `ReadOnly` that anything has happened at all when the checkbox is toggled.